Jenkinsfile

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env groovy
+
+pipeline {
+  agent {
+    label 'huge && bionic && x86_64 && !xenial && !trusty'
+  }
+
```

Comment:
could add a global option section, like
```groovy
    options {
        timeout(time: 1, unit: 'HOURS') 
    }
```

see https://www.jenkins.io/doc/book/pipeline/syntax/#options

or just for a stage, like
```groovy
stage('Prepare) {
  options {
    timeout(time: 1, unit: 'HOURS')
  }
  steps {
    # ....
  }
}
```

Reply:
I can add per-stage time-outs too. With this PR I just wanted to ensure feature parity with the existing job.

Perhaps when get some measures from max time elapsed by every stage on the different hosts, we can come up with some per-stage timeouts?